python/lsst/daf/butler/registry/_registry.py

```diff
@@ -497,6 +497,32 @@ def registerDatasetType(self, datasetType: DatasetType) -> bool:
         _, inserted = self._datasets.register(datasetType)
         return inserted
 
+    def pruneDatasetType(self, name: str) -> None:
+        """
+        Prune the named `DatasetType` from the registry.
+
+        .. warning::
+
+            Registry caches the dataset type definitions. This means that
+            deleting the dataset type definition may result in unexpected
+            behavior from other butler processes that are active that have
+            not seen the deletion.
+
+        Parameters
+        ----------
+        name : `str`
+            Name of the type to be removed.
+
+        Raises
+        ------
+        KeyError
+            Raised if the dataset type is not registered.
+        sqlalchemy.exc.IntegrityError
```

Comment:
I wish we had our own exception class for this sort of error instead of making high-level interface expose low-level/implementation exceptions,

Reply:
Maybe I should make one. I was pondering what to reraise although I guess we need a base class of RegistryDatabaseError and a subclass of RegistryIntegrityError -- trivial to add of course.

Reply:
It looks like `OrphanedRecordError` is already exactly the error I want.